tracing/src/main/java/com/palantir/remoting1/tracing/TracingAwareCallable.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting1.tracing;
+
+import java.util.Deque;
+import java.util.concurrent.Callable;
+
+public final class TracingAwareCallable<V> implements Callable<V> {
+
+    private final Callable<V> delegate;
+    private final Deque<TraceState> state;
+
+    public TracingAwareCallable(Callable<V> delegate) {
+        this.delegate = delegate;
+        this.state = Traces.getCopyOfState();
```

Comment:
utility class mentioned in previous comment would make it more apparent how to do the wrapping and more obvious that the state is being injected. The main thing is that a caller that needs to wrap their `Callable<V>` submission should be able to do so without having to know about how to grab present state.